apps/common/utils/format.tsx

```diff
@@ -0,0 +1,228 @@
+	options.displayDigits = assertValidNumber(options?.displayDigits, 0, 'displayDigits');
+
+	/* 🔵 - Yearn Finance *************************************************************************
+	** We need to ensure that minimumFractionDigits is a valid number. It can be any positive
+	** integer between 0 and 18. If the value is invalid, we display a warning and set the value to
+	** the default value (2 or 18).
+	**********************************************************************************************/
+	options.minimumFractionDigits = assertValidNumber(options?.minimumFractionDigits, 2, 'minimumFractionDigits');
+
+	/* 🔵 - Yearn Finance *************************************************************************
+	** We need to ensure that maximumFractionDigits is a valid number. It can be any positive
+	** integer between 0 and 18. If the value is invalid, we display a warning and set the value to
+	** the default value (2 or 18).
+	**********************************************************************************************/
+	options.maximumFractionDigits = assertValidNumber(options?.maximumFractionDigits, 2, 'maximumFractionDigits');
+
+	/* 🔵 - Yearn Finance *************************************************************************
+	** We need to ensure that maximumFractionDigits is always bigger or equal to
+	** minimumFractionDigits, otherwise we set them as equal.
+	**********************************************************************************************/
+	if (options.maximumFractionDigits < options.minimumFractionDigits) {
+		options.maximumFractionDigits = options.minimumFractionDigits;
+	}
+
+	if (options.shouldDisplaySymbol === undefined) {
+		options.shouldDisplaySymbol = true;
+	}
+	if (options.shouldCompactValue === undefined) {
+		options.shouldCompactValue = true;
+	}
+
+	return options;
+}
+function formatLocalAmount(
+	amount: number,
+	decimals: number,
+	symbol: string,
+	options: TAmountOptions
+): string {
+	/* 🔵 - Yearn Finance *************************************************************************
+	** Define the normalized elements.
+	** We use a few tricks here to get the benefits of the intl package and correct formating no
+	** matter the provided local.
+	** - Fallback formatting is set to `fr-FR`
+	** - If symbol is USD, then we will display as `123,79 $` or `$123.79` (US)
+	** - If smbol is percent, then we will display as `12 %` or `12%` (US)
+	** - If symbol is any other token, we will display as `123,79 USDC` or `USDC 123.79` (US)
+	**********************************************************************************************/
+	let locale = 'fr-FR';
+	if (typeof(navigator) !== 'undefined') {
+		locale = navigator.language || 'fr-FR';
+	}
+	const {shouldDisplaySymbol, shouldCompactValue, ...rest} = options;
+	const intlOptions: Intl.NumberFormatOptions = rest;
+	let isPercent = false;
+	if (symbol && symbol !== '' && shouldDisplaySymbol) {
```

Comment:
We can simplify this as empty string is falsey
```suggestion
	if (symbol && shouldDisplaySymbol) {
```